linalg/linalg.hpp

```diff
@@ -29,6 +29,7 @@
 #include "solvers.hpp"
 #include "handle.hpp"
 #include "invariants.hpp"
+#include "constrained.hpp"
```

Comment:
I'd prefer `constraints.hpp`

Reply:
Do you want the object changed to `ConstraintsSolver` or just the filename?

Reply:
No, I like `ConstrainedSolver` :)

Reply:
This reminds me that there may be some confusion with `ConstrainedOperator`... not sure what to do about thta

Reply:
To throw my hat in the ring, I've never really liked the existing `ConstrainedOperator` name because most (well, at least me) people will assume it does this functionality.

Reply:
It has propagated through the code, but maybe it is not too late to rename it. What do you think will be a better name?

The description (in `linalg/operator.hpp`) is:

```c++
/** @brief Square Operator for imposing essential boundary conditions using only
    the action, Mult(), of a given unconstrained Operator.

    Square operator constrained by fixing certain entries in the solution to
    given "essential boundary condition" values. This class is used by the
    general, matrix-free system formulation of Operator::FormLinearSystem. */
class ConstrainedOperator : public Operator
```

Reply:
Hmm, I think the problem comes down to the fact that a solver is an operator, so semantically a `ConstrainedSolver` should be equivalent to a `ConstrainedOperator` (which it isn't). Maybe `FixedBoundaryOperator`? That's clunky, I'll think about this some more.

Reply:
Maybe we can extend `ConstrainedOperator` to support more general constraints, e.g. with different constructors. Then it represents the action of the forward constrained operator and the `ConstrainedSolver` class is a base class for solvers for `ConstrainedOperator`s.

Reply:
Any thoughts on if the file should be named `constrained_solver.hpp` to avoid confusion?